worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ReducePartitionDataWriter.java

```diff
@@ -24,60 +24,33 @@
 import org.slf4j.LoggerFactory;
 
 import org.apache.celeborn.common.CelebornConf;
-import org.apache.celeborn.common.meta.DiskFileInfo;
 import org.apache.celeborn.common.meta.ReduceFileMeta;
 import org.apache.celeborn.common.metrics.source.AbstractSource;
-import org.apache.celeborn.common.protocol.PartitionSplitMode;
-import org.apache.celeborn.common.protocol.PartitionType;
 
 /*
  * reduce partition file writer, it will create chunk index
  */
 public final class ReducePartitionDataWriter extends PartitionDataWriter {
   private static final Logger logger = LoggerFactory.getLogger(ReducePartitionDataWriter.class);
 
-  private long nextBoundary;
-  private final long shuffleChunkSize;
-
   public ReducePartitionDataWriter(
       StorageManager storageManager,
-      DiskFileInfo diskFileInfo,
-      Flusher flusher,
       AbstractSource workerSource,
       CelebornConf conf,
       DeviceMonitor deviceMonitor,
-      long splitThreshold,
-      PartitionSplitMode splitMode,
-      boolean rangeReadFilter,
-      String shuffleKey)
+      PartitionDataWriterContext writerContext)
       throws IOException {
-    super(
-        storageManager,
-        diskFileInfo,
-        flusher,
-        workerSource,
-        conf,
-        deviceMonitor,
-        splitThreshold,
-        splitMode,
-        PartitionType.REDUCE,
-        rangeReadFilter,
-        shuffleKey);
-    this.shuffleChunkSize = conf.shuffleChunkSize();
-    this.nextBoundary = this.shuffleChunkSize;
-  }
-
-  @Override
-  protected void flush(boolean finalFlush) throws IOException {
-    super.flush(finalFlush);
-    maybeSetChunkOffsets(finalFlush);
+    super(storageManager, workerSource, conf, deviceMonitor, writerContext, true);
   }
 
-  private void maybeSetChunkOffsets(boolean forceSet) {
-    long bytesFlushed = diskFileInfo.getFileLength();
-    if (bytesFlushed >= nextBoundary || forceSet) {
-      ((ReduceFileMeta) diskFileInfo.getFileMeta()).addChunkOffset(bytesFlushed);
-      nextBoundary = bytesFlushed + shuffleChunkSize;
+  private void updateLastChunkOffset() {
```

Comment:
Shall we add a new method such as getCurrentFIleInfo, then we can probably improve updateLastChunkOffset and isChunkOffsetValid